pyfstat/core.py

```diff
@@ -962,13 +992,66 @@ def get_fullycoherent_twoF(
         tstart, tend: int or None
             GPS times to restrict the range of data used.
             If None: falls back to self.minStartTime and self.maxStartTime.
-            If outside those: auto-truncated.
+            This is only passed on to `self.get_transient_detstat()`,
+            i.e. only used if `self.transientWindowType` is set.
 
         Returns
         -------
         stat: float
             A single value of the detection statistic (twoF or log10BSGL)
             at the input parameter values.
+            Also stored as `self.twoF` or `self.log10BSGL`.
+        """
+        self.get_fullycoherent_twoF(
+            F0, F1, F2, Alpha, Delta, asini, period, ecc, tp, argp
+        )
+        if not self.transientWindowType:
+            if self.BSGL is False:
+                return self.twoF
+            self.get_fullycoherent_single_IFO_twoFs()
+            self.get_fullycoherent_log10BSGL()
+            return self.log10BSGL
+        self.get_transient_maxTwoFstat(tstart, tend)
+        if self.BSGL is False:
+            return self.maxTwoF
+        else:
+            return self.get_transient_log10BSGL()
```

Comment:
I find this block kind of convoluted. Reminder to check futher.

Reply:
It's the "return early" philosophy which I also personally find harder to read than a fully hierarchical if-else construct, but I decided to keep it logically in that same structure as it was before.